Mathlib/Data/Nat/Fib/Lemmas.lean

```diff
@@ -0,0 +1,79 @@
+/-
+Copyright (c) 2025 Monica Omar. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Monica Omar
+-/
+import Mathlib.Data.Nat.Fib.Basic
+import Mathlib.LinearAlgebra.Matrix.Notation
+import Mathlib.LinearAlgebra.Matrix.Determinant.Basic
+
+/-!
+# Cassini and Catalan identities for the Fibonacci numbers
+
+Cassini's identity states that for nonzero `n`, `fib (n + 1) * fib (n - 1) - fib n ^ 2` is equal
+to `(-1) ^ n`. And Catalan's identity states that for nonzero `x` and `a`, we get
+`fib (x + a) ^ 2 - fib x * fib (x + 2 * a) = (-1) ^ x * fib a ^ 2`.
+-/
+
+namespace Nat
+
+/-- For any nonzero `n`, the matrix `[fib (n + 1), fib n; fib n, fib (n - 1)]` is equal to
+`[1, 1; 1, 0] ^ n`. -/
+lemma fib_matrix_eq : ∀ {n : ℕ}, n ≠ 0 →
+    !![fib (n + 1), fib n; fib n, fib (n - 1)] = !![1, 1; 1, 0] ^ n
+  | 1, _ => by simp
+  | n + 2, _ => by
+    rw [pow_succ, ← fib_matrix_eq (add_one_ne_zero n)]
+    simp [fib_add_two, ← add_assoc, add_rotate, add_comm (fib n)]
+
+/-- **Cassini's identity**: for nonzero `n`, `fib (n + 1) * fib (n - 1) - fib n ^ 2 = (-1) ^ n`. -/
+lemma fib_succ_mul_fib_pred_sub_fib_sq {n : ℕ} (hn : n ≠ 0) :
+    (fib (n + 1) * fib (n - 1) - fib n ^ 2 : ℤ) = (-1) ^ n :=
```

Comment:
Likewise:
```suggestion
lemma fib_add_two_mul_fib_sub_fib_succ_sq (n : ℕ) :
    (fib (n + 2) * fib n - fib (n + 1) ^ 2 : ℤ) = (-1) ^ (n + 1) :=
```
Integer subtraction is unfortunate, but I don't think there's a way to avoid it here.